lib/DOEKBase/DataAPI/annotation/genome_annotation/ClientAPI.pm

```diff
@@ -96,28 +96,79 @@ sub get_feature_ids {
 
     my %args=@_;
 
-#    print Dumper(@args);
-
-#filters={'type_list':['CDS','mRNA'],'region_list':[],'function_list':[],'alias_list':[]}
-
-#      $self->{'client'}->get_feature_ids($self->{'token'},$self->{'ref'}, {type_list=>['CDS'],'region_list'=>[],'function_list'=>[],'alias_list'=>[]},{});
-
     my $converted_filters = DOEKBase::DataAPI::annotation::genome_annotation::Feature_id_filters->new();
     foreach my $filter (keys %{$args{'filters'}})
     {
         $converted_filters->{$filter}=$args{'filters'}{$filter};
     }
+    if ($args{'filters'}{'region_list'})
+    {
+        my @regions = map { DOEKBase::DataAPI::annotation::genome_annotation::Region->new($_) } @{$args{'filters'}{'region_list'}};
+        $converted_filters->{'region_list'} = \@regions;
+    }
+
+    my $group_by = $args{'group_by'} || 'type';
 
     my $result = try {
-#      $self->{'client'}->get_feature_ids($self->{'token'},$self->{'ref'}, $converted_filters,{});
-      $self->{'client'}->get_feature_ids($self->{'token'},$self->{'ref'},$converted_filters);
+      $self->{'client'}->get_feature_ids($self->{'token'},$self->{'ref'},$converted_filters,$group_by);
     } catch {
       no warnings 'uninitialized';
       no strict 'refs';
       confess "$_ Exception thrown by get_feature_ids: code " . $_->{'code'} . ' message ' . $_->{'message'};
     };
 
-    return $result;
+    my $final_result = {};
+
+    foreach my $group_key (qw(by_type by_function by_alias))
+    {
+        foreach my $type (keys %{$result->{$group_key}})
+        {
+            push @{$final_result->{$group_key}{$type}}, map { [ $_->feature_type(),$_->feature_id() ] } @{$result->{$group_key}{$type}};
+        }
+    }
+
+    # filter by region is a different structure
+    foreach my $contig (keys %{$result->{'by_region'}})
+    {
+        foreach my $strand (keys %{$result->{'by_region'}{$contig}})
+        {
+            foreach my $region (keys %{$result->{'by_region'}{$contig}{$strand}})
+            {
+                push @{$final_result->{'by_region'}{$contig}{$strand}{$region}}, map { [ $_->feature_type(),$_->feature_id() ] } @{$result->{'by_region'}{$contig}{$strand}{$region}};
+            }
+        }
+    }
+
+    # only return relevant group_by key
+    return {"by_$group_by",$final_result->{"by_$group_by"} };
+
   }
 
+sub get_features {
```

Comment:
This new get_features() code is probably not compatible with rejecting PR 133.  I think the only thing that would be needed is to remove it completely, and add get_features back to the list of autogenerated methods.